src/server/snapshot.cc

```diff
@@ -238,13 +238,26 @@ bool SliceSnapshot::BucketSaveCb(PrimeIterator it) {
   ++stats_.savecb_calls;
 
   uint64_t v = it.GetVersion();
-  if (v >= snapshot_version_) {
-    // either has been already serialized or added after snapshotting started.
-    DVLOG(3) << "Skipped " << it.segment_id() << ":" << it.bucket_id() << ":" << it.slot_id()
-             << " at " << v;
-    ++stats_.skipped;
+  auto check = [&](auto v) {
+    if (v >= snapshot_version_) {
+      // either has been already serialized or added after snapshotting started.
+      DVLOG(3) << "Skipped " << it.segment_id() << ":" << it.bucket_id() << ":" << it.slot_id()
+               << " at " << v;
+      ++stats_.skipped;
+      return true;
+    }
+    return false;
+  };
+
+  if (check(v))
     return false;
+
+  while (bucket_ser_in_progress_) {
```

Comment:
This will block all operations on the thread, even actions on different buckets. Effectively this means `if the bucket is not serialized and there is serialization in progress. block`

Reply:
definitely do not actively poll. this will consume 100% CPU, causing alerts and such, especially on huge entries...